_drafts/2017-06-07-checkers.md

```diff
@@ -0,0 +1,114 @@
+authors:
+  - name: Noam Ross
+    url: http://twitter.com/noamross
+  - name: Alice Daish
+    url: http://twitter.com/alice_data
+  - name: Laura DeCicco
+    url: http://twitter.com/DeCiccoDonk 
+  - name: Nistara Randhawa
+    url: http://twitter.com/nistara_r
+  - name: Jennifer Thompson
+    url: http://twitter.com/jent103
+  - name: Nick Tierney
+    url: https://twitter.com/nj_tierney
+categories:
+  - blog
+tags:
+  - unconf
+  - testing
+  - packages
+  - software
+  - community
+  - meetings
+---
+
+Two years ago at [\#runconf15], there was a great discussion about best practices for organizing R-based analysis projects that yielded a [nice guidance document] describing *research compendia*. Compendia, as we described them, were minimal products of reproducible research, using parts of R package structure to organize the inputs, analyses, and outputs of research projects.
+ 
+Since then, we've seen more examples and models of research compendia emerge (the organization of such projects is [something of an obsession] for some of the community.) In parallel, there's been much progress on a number of fronts with R *packages*: ROpenSci's [package review process] has expanded and we've worked out many kinks. Infrastructure for automated testing of package code has been developed and field tested. So at [\#runconf17](unconf17.ropensci.org), we wanted to see how much of this progress in review, testing, and automation could apply to research compendia.
+ 
+It turns out there's a *lot* to do here, and a lot of interest! We put up a proposal as a [GitHub issue]; before the unconf even began, the thread had over 50 posts and the discussion had yielded two design documents led by Hadley Wickham on [research compendium organization] and [automated build systems]. There were easily four or five projects wrapped up in the proposal.
+ 
+The thread also revealed many schools of thought. As [Nick] put it, "The problem with [opinionated analysis development] is that everyone has an opinion." We never reached consensus on basic issues like directory structure and build systems.
+
+![](http://imgur.com/RpjyfL8.jpeg)
+ 
+In the face of a beast of a topic, and so many unresolved decisions, we decided to tackle a modest slice - a guide for compendium review, and a first pass at a package for automated checks of various best practices.
+
+# Reviewing a research compendium
+
+How does one do peer review for a research compendium in the face of so much heterogeneity in types and styles of analysis?  We spent the better part of our first day brainstorming everything that one might review, from the meta-topics ("Is there a clear question?") to  mintuae ("Are `library()` calls at the top of the script?"), and mapping these out on an ever more data-dense whiteboard:
+
+![Nick, Jennifer, Molly and Alice discussing analysis best practises](http://imgur.com/vb4E3JV.jpeg)
+ _Nick, Jennifer, Molly and Alice discussing analysis best practises_
```

Comment:
In various parts of this section,  s/practises/practices and same for s/practise/practice.